Rules/CheckCmdletAvailable.cs

```diff
@@ -0,0 +1,127 @@
+﻿//
+// Copyright (c) Microsoft Corporation.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Management.Automation.Language;
+using System.Reflection;
+using Microsoft.Windows.Powershell.ScriptAnalyzer.Generic;
+using System.ComponentModel.Composition;
+using System.Globalization;
+
+namespace Microsoft.Windows.Powershell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// CheckCmdletAvailable: Check if cmdlet is available on NanoServer.
+    /// </summary>
+    [Export(typeof(IScriptRule))]
+    public class CheckCmdletAvailable : IScriptRule
+    {
+        /// <summary>
+        /// AnalyzeScript: Check if cmdlet is available on NanoServer.
+        /// </summary>
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            IEnumerable<Ast> cmdletAsts = ast.FindAll(testAst => testAst is CommandAst, true);
+            if (cmdletAsts.Count() != 0)
+            {
+                List<string> availableCmdlets = GetAvailableCommands();
+                foreach (CommandAst cmdletAst in cmdletAsts)
+                {
+                    //Check if the command name is in the whitelist.
+                    string cmdletName = cmdletAst.GetCommandName();
+                    if (!availableCmdlets.Any( s=>s.Equals(cmdletName,StringComparison.OrdinalIgnoreCase))) 
+                    {
+                        yield return new DiagnosticRecord(string.Format(CultureInfo.CurrentCulture, Strings.CmdletNotAvailableOnNanoError, cmdletName),
+    cmdletAst.Extent, GetName(), DiagnosticSeverity.Warning, fileName);
+                    }
+                }
+            }
+            
+        }
+
+        /// <summary>
+        /// GetAvailableCommands: Retrieve the list of commands that are available from the white list.
+        /// </summary>
+        /// <returns></returns>
+        public List<string> GetAvailableCommands()
+        {
+            List<string> availableCmdlets = new List<string>();
+            string path = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+            string[] lines = File.ReadAllLines(string.Format(CultureInfo.CurrentCulture, path+"../../Rules/CommandOnNano.txt"));
+            foreach (string line in lines)
+            {
+                availableCmdlets.Add(line);
```

Comment:
Ideally, we should only process the file this once. Maybe add a check to see whether it has been run before?